object-cache.php

```diff
@@ -1532,9 +1532,21 @@ public function getAllOptions() {
 
 		// getMulti returns a map of `[ cache_key => value ]` but we need to
 		// return a map of `[ option_name => value ]`
-		$data = array_combine( $keys, $data );
+		// Merging data from cache server and keys
+                // We cannot simply do a 'array_combine' on the returned $data with $keys since in a multi nodes environment, it might not return the data in order as the order of the keys passed in.
+                // For example: keys = array("a", "b", "c"), data returned = array("a" => "valueA", "c" => "valueC", "b" => "valueB")
+                // An array_combine would return array("a" => "valueA", "b" => "valueC", "c" => "valueB") 
```

Comment:
Or could lead to a warning from array_combine in the case where one node has evicted some of the values and hence $keys and $data would be of different lengths.